pkg/quack/quack_test.go

```diff
@@ -98,3 +142,57 @@ func TestRequestHasAnnotation(t *testing.T) {
 	assert.False(t, withoutRequired, "Object without required annotation should return false")
 	assert.True(t, noAnnotation, "Specifying no required annotation should return true")
 }
+
+func TestGetDelims(t *testing.T) {
+	objectWithNoAnnotations := struct {
+		metav1.ObjectMeta `json:"metadata"`
+	}{}
+	objectWithSetDelimiters := struct {
+		metav1.ObjectMeta `json:"metadata"`
+	}{
+		ObjectMeta: metav1.ObjectMeta{
+			Annotations: map[string]string{
+				leftDelimAnnotation:  "[[",
+				rightDelimAnnotation: "]]",
+			},
+		},
+	}
+	objectWithEmptyDelimiters := struct {
+		metav1.ObjectMeta `json:"metadata"`
+	}{
+		ObjectMeta: metav1.ObjectMeta{
+			Annotations: map[string]string{
+				leftDelimAnnotation:  "",
+				rightDelimAnnotation: "]]",
+			},
+		},
+	}
+
+	objectWithNoAnnotationsRaw, err := json.Marshal(objectWithNoAnnotations)
+	if err != nil {
+		assert.FailNowf(t, "jsonError", "Failed to marshal 'with no annotations' input: %v", err)
+	}
+	objectWithSetDelimitersRaw, err := json.Marshal(objectWithSetDelimiters)
+	if err != nil {
+		assert.FailNowf(t, "jsonError", "Failed to marshal 'with set delimeters' input: %v", err)
+	}
+	objectWithEmptyDelimitersRaw, err := json.Marshal(objectWithEmptyDelimiters)
+	if err != nil {
+		assert.FailNowf(t, "jsonError", "Failed to marshal 'with empty delimeter' input: %v", err)
+	}
+
+	withNoAnnotations, err := getDelims(objectWithNoAnnotationsRaw)
+	if err != nil {
+		assert.FailNowf(t, "methodError", "Error in getDelims: %v", err)
+	}
+	withSetDelimters, err := getDelims(objectWithSetDelimitersRaw)
+	if err != nil {
+		assert.FailNowf(t, "methodError", "Error in getDelims: %v", err)
+	}
+	withEmptyDelimeters, err := getDelims(objectWithEmptyDelimitersRaw)
```

Comment:
As discussed, this could use a test case for only left and only right delim in the annotations